src/unitxt/inference.py

```diff
@@ -73,6 +73,7 @@ def prepare(self):
             model=self.model_name, trust_remote_code=True, **model_args
         )
 
+    @InferenceEngine.verify_dataset
```

Comment:
Do we prefer to use decorator? It requires the writer of the inference engine to remember to add the decorator.
Maybe better have infer() that calls verify_dataset()  and then _infer() which will be implemented by each inference engine.

Reply:
Yup, it may be better